webpack/assets/javascripts/react_app/components/Layout/components/UserDropdowns.js

```diff
@@ -21,32 +22,31 @@ const UserDropdowns = ({
     >
       <NotificationContainer data={{ url: notificationUrl }} />
     </NavItem>
-    {!!user &&
-      !!user.current_user && (
-        <NavDropdown componentClass="li" id="account_menu">
-          <Dropdown.Toggle useAnchor className="nav-item-iconic">
-            <Icon type="fa" name="user avatar small" />
-            {user.current_user.user.firstname} {user.current_user.user.lastname}
-          </Dropdown.Toggle>
-          <Dropdown.Menu>
-            {user.user_dropdown[0].children.map((item, i) =>
-                (item.type === 'divider' ? (
-                  <MenuItem key={i} divider />
-                ) : (
-                  <MenuItem
-                    key={i}
-                    className="user_menuitem"
-                    href={item.url}
-                    onClick={() => {
-                      changeActiveMenu({ title: 'User' });
-                    }}
-                  >
-                    {__(item.name)}
-                  </MenuItem>
-                )))}
-          </Dropdown.Menu>
-        </NavDropdown>
-      )}
+    {user && currentUser && (
```

Comment:
checking `currentUser` isn't enough?

Reply:
Technically we could just check currentUser since we know it is coming from the `user` object in the parent component. But in the isolated space of this component, these are two distinct inputs and both are being called in the scope that this check protects. I don't think it is a good idea to assume knowledge of the parent component here, but I am open on how to improve this